src/routes/api/products.routes.js

```diff
@@ -42,11 +42,9 @@ route.patch(
   checkRole(['seller']),
   updateProductAvailability
 );
-route.patch('/:id', extractToken, checkRole(['seller']),checkIfPasswordIsExpired, updateProduct);
-route.delete('/:id', extractToken, checkRole(['seller']),checkIfPasswordIsExpired, deleteProduct);
-route.get('/:id/product-wishes',extractToken,getWishesPerProduct)
-route.patch('/:id', extractToken, checkRole(['seller']), updateProduct);
-route.delete('/:id', extractToken, checkRole(['seller']), deleteProduct);
-route.post('/:id/reviews',extractToken,checkRole(['buyer']),validateReviewProduct,createProductReview)
+route.patch('/:id', extractToken, checkRole(['seller']), uploadArray('image'), updateProduct, checkIfPasswordIsExpired);
```

Comment:
A middleware should come before controller; change this checkIfPasswordIsExpired middleware position